lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -7273,9 +7273,23 @@ class DecomposeAtenNativeLayerNormOp
     Value inputMean = rewriter.create<AtenMeanDimOp>(
         loc, reducedTy, op.getInput(), reduceDimList, cstTrue, none);
 
+    Value cstFalse = rewriter.create<Torch::ConstantBoolOp>(loc, false);
+    auto castToTargetDType = [&](Value tensor, Type targetDtype) -> Value {
+      auto tensorTy = cast<ValueTensorType>(tensor.getType());
+      if (tensorTy.getDtype() == targetDtype)
+        return tensor;
+      auto castedType = ValueTensorType::get(rewriter.getContext(),
+                                             tensorTy.getSizes(), targetDtype);
+      Value dtypeValue =
+          Torch::getDtypeIntValueForType(rewriter, loc, targetDtype);
+      return rewriter.create<Torch::AtenToDtypeOp>(
+          loc, castedType, tensor, dtypeValue, cstFalse, cstFalse, none);
+    };
+
+    Value inputMeanCasted = castToTargetDType(inputMean, inputTy.getDtype());
```

Comment:
Instead, you can do:
```suggestion
    Value inputMeanCasted = convertTensorToDtype(rewriter, loc, inputMean, inputTy.getDtype());
```

Reply:
Thanks for pointing this out! I did not know about this utility.

Done.